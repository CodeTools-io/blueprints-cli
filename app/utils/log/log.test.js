const log = require('./log')

describe('log', () => {
  test('can log standard output', () => {
    log.clear()
    log('first')
    log('second')
    log('third')
    const result = log.output()

    expect(result).toMatchSnapshot()
  })

  test('can log warning output', () => {
    log.clear()
    log.warning('first warning')
    log.warning('second warning')
    log.warning('third warning')
    const result = log.output()

    expect(result).toMatchSnapshot()
  })

  test('can log error output', () => {
    expect(() => {
      log.error('first error')
    }).toThrow('first error')
  })
})
