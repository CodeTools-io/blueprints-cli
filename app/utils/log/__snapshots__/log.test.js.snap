// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log can log standard output 1`] = `
"first
second
third"
`;

exports[`log can log warning output 1`] = `
"⚠️ first warning
⚠️ second warning
⚠️ third warning"
`;
